index.ts

```diff
@@ -0,0 +1,139 @@
+import express from "express";
+import {exec} from "child_process";
+import crypto from "crypto";
+import dotenv from "dotenv";
+
+dotenv.config();
+
+const app = express();
+const port = process.env.PORT || '3333';
+const secret = process.env.GITHUB_SECRET;
+if(!secret) {
+  console.error("Error: GITHUB_SECRET environment variable is not set.");
+  process.exit(1);
+};
+
+let isDocumentationWebsiteUpdated = false;
+let isMindmapUpdated = false;
+let contributorsBuildRequired = false;
+
+let documentationWebsiteBuildTime = 0;
+let mindmapBuildTime = 0;
+let contributorsBuildTime = 0;
+
+app.use(express.json());
+
+app.post("/webhook", async (req, res) => {
+  console.log("req receieved");
+  const signature = req.headers["x-hub-signature"] as string;
```

Comment:
what happens if there isn't a header?

Reply:
Are you referring to the ending of the signature variable?

Reply:
no, I am refering to the case, that there is a request without the x-hub-signature header